Automattic/jetpack
# JS Components: Convert layout (Col, Container) components to TypeScript

#### Changes proposed in this Pull Request:
* This PR converts `Col` and `Container` components to TypeScript.
* It also converts the Layout stories to TS

To see the actual changes please check d9f3a3110802beeba0c799f18b62118745d3be61 and 347861324db4e74545db4992a3beb498954c9eca

#### Other information:


#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the Layout stories work as expected
* Check the consumers of `Col` and `Container` and related components and confirm that they work as expected
* Check for any regression

<img width="876" alt="Screenshot 2022-07-29 at 12 46 46 PM" src="https://user-images.githubusercontent.com/18226415/181704436-c5e67542-796d-4517-9229-f06366b948e4.png">

<img width="893" alt="Screenshot 2022-07-29 at 12 45 46 PM" src="https://user-images.githubusercontent.com/18226415/181704257-eff55fe1-83fa-4119-bb63-33402bf6125e.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202276314264006
